File: 2244-number-of-laser-beams-in-a-bank/2244-number-of-laser-beams-in-a-bank.py
class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        lst = []
        for i in bank:
            c = i.count('1')
            if c:
                lst.append(c)

        c = 0
        for i in range(1, len(lst)):
            c += lst[i - 1] * lst[i]
        
        return c

File: 2244-number-of-laser-beams-in-a-bank/2244-number-of-laser-beams-in-a-bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        vector<int> b;
        for(auto &s : bank) {
            int c = 0;
            for(auto &ch : s) if(ch == '1') c++;
            if(c != 0) b.push_back(c);
        }
        int ans = 0;
        for(int i = 1; i < b.size(); i++) {
            ans += b[i] * b[i - 1];
        }
        return ans;
    }
};